### PR TITLE
add fields to default transformation

### PR DESCRIPTION
**Pull request by acoburn**
See: https://jira.duraspace.org/browse/FCREPO-1968

### CODE REVIEW

**Comment by ajs6f** [comment]
Wow, this is a load of stuff. What's the use case for this? My understanding of the default transform is that it's not supposed to be useful, it's just supposed to prevent anything from erroring-out.

**Comment by acoburn** [comment]
@ajs6f the use case is this: at just about every Fedora workshop I've ever been to, there is the recurring complaint that "Fedora has no search capability". The _only_ purpose of this PR is to add more fields to the default transform so that something akin to an "out of the box" search capability (when paired with `fcrepo-indexing-solr`) can be offered up without enshrining this in the core of Fedora. The insistence that a sidecar triplestore can be your search engine seems to be inadequate, so the idea here is simply to make a solr-based sidecar search engine easier to build.

That said, I'm also open to paring down this list of properties. At a minimum, it should contain all of the `fedora:` namespaced properties and some relevant `rdf/rdfs` properties. The `skos`, `dc/dcterms`, `ore`, `acl` and `pcdm` properties are just for convenience.

**Comment by ajs6f** [comment]
Then I would take this list of fields and put it in a new prepackaged transform called something to the purpose. I think this badly confuses the semantics of "default".

**Comment by acoburn** [comment]
That works for me.

**Comment by escowles** [comment]
Having a very minimal `default` transform and this is a `deluxe` transform sounds good to me.

**Comment by ajs6f** [comment]
`deluxe` is a good term, although I would also like `so, you're too lazy to bother actually configuring the powerful repository software in which you've invested and would prefer that it all somehow "just work"?`.

**Comment by acoburn** [comment]
URL-encoded, that would be `so%2C%20you%27re%20too%20lazy%20to%20bother%20actually%20configuring%20the%20powerful%20repository%20software%20in%20which%20you%27ve%20invested%20and%20would%20prefer%20that%20it%20all%20somehow%20%22just%20works%22%3F`. 

I like it.

**Comment by acoburn** [comment]
I left `default` as is it and added these fields to a `deluxe` transformation.

File: src/main/java/org/fcrepo/transform/http/FedoraTransform.java [hunk 1]
```diff
@@ -119,16 +122,27 @@ public void setUpRepositoryConfiguration() throws RepositoryException, IOExcepti
...
-                final Node baseConfig = node.addNode(NT_BASE, NT_FILE);
-                jcrTools.uploadFile(internalSession, baseConfig.getPath(), getClass().getResourceAsStream(
-                        "/ldpath/default/nt_base_ldpath_program.txt"));
-            }
+
+            final Map<String, String> transformations = of(
+                    "default", "/ldpath/default/ldpath_program.txt",
+                    "deluxe", "/ldpath/deluxe/ldpath_program.txt");
+
+            transformations.entrySet().forEach(entry -> {
```
Comment (ajs6f) [inline src/main/java/org/fcrepo/transform/http/FedoraTransform.java]: Why not just `Map::forEach` here? You're just building up a set of `Entry` and then breaking it back down into keys and values?

Comment (acoburn) [inline src/main/java/org/fcrepo/transform/http/FedoraTransform.java]: Good point. This is now fixed.

**Comment by awoods** [comment]
@acoburn, will you have an associated PR for fcrepo-camel-toolbox that invokes the `deluxe` transform?

**Comment by acoburn** [comment]
@awoods no PR needed, since that's just a configuration value. Unless you're suggesting changing the default configuration from 'default' to 'deluxe'.

**Comment by awoods** [comment]
Sounds good, @acoburn. No additional PR needed... just some eventual documentation to help move interested users in the `deluxe` direction.

**Comment by awoods** [comment]
Resolved with commit to DEV: https://github.com/fcrepo4-exts/fcrepo-transform/commit/003d7b3dade6fe2007c1b9f3210157b0f32cfe08

**Pull request closed without merging**